### PR TITLE
New arm sim

### PR DESCRIPTION
**Pull request by ssarahyao**
New arm simulation is very cool and good.

### CODE REVIEW

File: 2024 Code/src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 1]
```diff
@@ -70,6 +69,8 @@ public void armPID(double targetAngle) {
     double FFoutput = feedforward.calculate(m_angle, velocity, acceleration);
 
     setArmVoltage(PIDoutput + FFoutput);
+
+    m_targetAngle = targetAngle;
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/ArmSubsystem.java:73]: why is this being set here? i'ts already set above

File: 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java [hunk 2]
```diff
@@ -53,21 +67,43 @@ public ArmSimulation() {
...
         REVPhysicsSim.getInstance().addSparkMax(m_arm, DCMotor.getNEO(1));
+        pidSim = new CanSparkMaxPidSim();
     }
 
     @Override
     public void simulationPeriodic() {
-        double motor_volts = m_arm.getAppliedOutput() * m_arm.getBusVoltage();
+        // motor_volts = m_arm.getAppliedOutput() * m_arm.getBusVoltage();
+
+        // System.out.println("this function is running");
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java:83]: can you clean up all these comments

File: 2024 Code/src/main/java/frc/robot/subsystems/simulation/Shooter.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 public class Shooter extends ShooterSubsystem {
     Mechanism2d m_mech = new Mechanism2d(3, 3);
     MechanismRoot2d m_root = m_mech.getRoot("chassis", 2, 2);
-    MechanismLigament2d m_wheel_ligament;
+    public static MechanismLigament2d m_wheel_ligament;
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/simulation/Shooter.java:22]: why touch shooter?

File: 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java [hunk 4]
```diff
@@ -53,21 +67,43 @@ public ArmSimulation() {
...
+
+        get_angle = m_armSim.getAngleRads() * (180 / Math.PI);
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java:85]: https://www.scaler.com/topics/java-math-todegrees/

File: 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java [hunk 5]
```diff
@@ -53,21 +67,43 @@ public ArmSimulation() {
...
+        // * center of mass (r / 2) gives the torque
+        double gravity_torque_comp = (Math.cos(get_angle * (Math.PI / 180)) * 9.8 * 7) * 0.32775;
+
+        double ff = gravity_torque_comp * kG;
+
+        // double ff = (kcos * Math.cos(get_angle * (Math.PI / 180)) +
+        // (Math.signum(Math.cos(get_angle * (Math.PI / 180))) * ksin * 0.65
+        // * Math.sin((Math.PI / 180) * get_angle)));
+        System.out.println(ff);
+        motor_volts = pidSim.runPid(0.8, 0.0, 0.0,
```
Comment (gurellia53) [inline 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java:98]: I assume you will change this to use the calibrated values?

Comment (ssarahyao) [reply 2024 Code/src/main/java/frc/robot/subsystems/simulation/ArmSimulation.java:89]: si senor

**Review by TrentAgnos — APPROVED** [review]

**Pull request merged**